packages/stdlib/test/test-async.js

```diff
@@ -0,0 +1,50 @@
+// @ts-check
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava';
+import { asyncAllTruthies, asyncWhile } from '../src/ring1/async';
+
+test('asyncAllTruthies - for-await-of', async t => {
+  const results = [];
+  let state = 4;
+  const thunk = async () => {
+    const cur = await Promise.resolve(state);
+    state -= 1;
+    return cur;
+  };
+  for await (const el of asyncAllTruthies(thunk)) {
```

Comment:
It’ll be hard to judge how well this name will stand the test of time until we see the breadth of the palette the stdlib will need, so I’ll just recommend keeping an eye on this.

My thought process for this name goes like:

1. `all` implies returning an array, so that’s good
2. `truthies` is weird and should be implied by `all`
3. `async` is fine
4. nothing in this name says anything about discarding the suffix. Python’s itertools sets up a precedent for names like `take` and `drop` followed by `while` or `until` for this purpose.
5. But `while` and `until` imply a guard, but the guard is reflexive in this case. Maybe this function implies a default guard function that is reflexive.

Reply:
I made an attempt to fix and generalise these, following your point 5 (second argument guard function, with a default that is reflexive):

- `asyncWhile -> asyncDoWhile`
- `asyncAllTruthies -> asyncGenerateWhile`

also attn: @erights.